sweet/harnesses/cockroachdb.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2024 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package harnesses
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"golang.org/x/benchmarks/sweet/common"
+	"golang.org/x/benchmarks/sweet/common/log"
+)
+
+type CockroachDB struct{}
+
+func (h CockroachDB) CheckPrerequisites() error {
+	return nil
+}
+
+func (h CockroachDB) Get(gcfg *common.GetConfig) error {
+	// Build against the latest stable release.
+	return gitShallowClone(
+		gcfg.SrcDir,
+		"https://github.com/cockroachdb/cockroach",
+		"release-24.1",
+	)
+}
+
+func (h CockroachDB) Build(cfg *common.Config, bcfg *common.BuildConfig) error {
+	// Build the cockroach binary.
+	// We do this by using the cockroach `dev` tool. The dev tool is a bazel
+	// wrapper normally used for building cockroach, but can also be to
+	// generate artifacts that can then be built by `go build`.
+
+	// Install bazel via bazelisk which is used by `dev`.
+	if err := cfg.GoTool().Do("", "install", "github.com/bazelbuild/bazelisk@latest"); err != nil {
+		return fmt.Errorf("error building bazelisk: %v", err)
+	}
+
+	// Configure the build env.
+	env := cfg.BuildEnv.Env
+	env = env.Prefix("PATH", filepath.Join(cfg.GoRoot, "bin")+":")
+	env = env.MustSet("GOROOT=" + cfg.GoRoot)
+
+	// Even though we aren't actually building anything with dev or bazel,
+	// the dev tool expects this set or else it won't let us do anything
+	// including generating the go files. Normally the `dev doctor` tool
+	// does this for us, but it doesn't have write permissions out of the
+	// box. It's easier to write it ourselves than to configure permissions
+	// and let it potentially change anything else unexpectedly.
+	bazelrc, err := os.Create(filepath.Join(bcfg.SrcDir, ".bazelrc.user"))
+	if err != nil {
+		return err
+	}
+	defer bazelrc.Close()
+	setting := []byte("build --config=dev")
+	_, err = bazelrc.Write(setting)
+	if err != nil {
+		return err
+	}
+
+	// Run dev doctor which will set up the environment for
+	// the dev tool to work properly.
+	cmd := exec.Command("./dev", "doctor")
+	cmd.Dir = bcfg.SrcDir
+	cmd.Env = env.Collapse()
+	log.TraceCommand(cmd, false)
+
+	// `dev doctor` is an interactive tool, pipe in input to
+	// decline all of optional settings it asks us about.
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return err
+	}
+	stdin.Write([]byte("n\nn\n\n"))
+	if err = cmd.Run(); err != nil {
+		return err
+	}
+
+	// Call `dev gen go` which will generate the Go code as well
+	// as the c-deps needed.
+	cmd = exec.Command("./dev", "gen", "go")
+	cmd.Dir = bcfg.SrcDir
+	cmd.Env = env.Collapse()
```

Comment:
Should we redirect `stdout/stderr` since the command may take a while and mislead the user, thinking that it's "hanging"?

```
cmd.Stdout = os.Stdout
cmd.Stderr = os.Stderr
```

Reply:
Done for the bazel run commands.